apps/web/src/util/navigation.js

```diff
@@ -195,7 +216,7 @@ export function widgetUrl ({ widget, rootPath, groupSlug: providedSlug, context
   } else if (widget.viewUser) {
     url = personUrl(widget.viewUser.id, groupSlug)
   } else if (widget.viewPost) {
-    url = postUrl(widget.viewPost.id, { groupSlug, context })
+    url = postUrl(widget.viewPost, { groupSlug, context })
```

Comment:
Did we double check that the ContextWidget queries on type Group are returning everything needed for the new postUrl, now that it depends on more than id? The query might be partial

Reply:
oops, this was supposed to stay as id. postUrl didnt actually end up changing. i messed with it at first and then just created the new primaryPostUrl function